### DISCUSSION TITLE
How i can create hash password and match password for different users

### DISCUSSION THREAD

**Discussion by amir7896** [deta/cloud-docs#466] [Help] [answered]
I have created the Deta  DB but i want to store the users password hash and while when user login then hash password should be compare with plan password i don't know how i could achieve this

**Comment by jnsougata** [comment]
1. What language you are using?
2. What did you mean by `plan password`?

**Reply by amir7896** [reply]
i'm using nodejs

**Comment by Sokhavuth** [comment, marked as answer]
You could use bcrypt that is independent from any kind of database. Here is an example of using bcrypt in Node.js:
```js
// npm install bcryptjs
const bcrypt = require("bcryptjs")

// to hash a password
const hashedPassword = bcrypt.hashSync("password", 8)
```

```js
const bcrypt = require("bcryptjs")

// to verify password
if(bcrypt.compareSync(req.body.password, user.hashedPassword)){
    ...
}else{
    ...   
}
```
Here is my website on Deta in which bcrypt is being used to hash and verify password: [https://khmerweb.deta.dev/](https://khmerweb.deta.dev/)

**Reply by amir7896** [reply]
Thanks for your response it save my time.
Is there any concept of models in Deta Like in mongodb (a user can create different models in single db like users, posts, reviews models in SocialDB).

**Reply by Sokhavuth** [reply]
In Deta, everything is collection or model if you like to call it. There is no concept of "database" as in MongoDB. So, the starting point in Deta database  is collection. Once you create a collection, you can insert document(s) in the same way as you do with MongoDB. However, you can create an empty database object to store collections as below:

```js
// npm install deta dotenv

const { Deta } = require("deta")
const dotenv = require('dotenv')
dotenv.config()

// connecting to your database
const deta = Deta(process.env.PROJECT_KEY)

// create an empty database object
const db = {}

// creating collections
db.users = deta.Base("users")
db.posts = deta.Base("posts")
db.categories = deta.Base("categories")

module.exports = db 
```

```js
async function createPost(req){
        const new_post = {
            title: req.body.title,
            content: req.body.content,
            categories: categories,
            thumb: req.body.thumb,
            date: req.body.datetime,
            videos: req.body.videos,
            author: req.myuserid,
        }

        await req.db.posts.put(new_post)
    }
```

**Reply by amir7896** [reply]
thank you for your precious response
